compiler/rustc_attr_parsing/src/parser.rs

```diff
@@ -100,10 +77,7 @@ impl<'a> PathParser<'a> {
 
 impl Display for PathParser<'_> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        match self {
-            PathParser::Ast(path) => write!(f, "{}", pprust::path_to_string(path)),
-            PathParser::Attr(attr_path) => write!(f, "{attr_path}"),
-        }
+        write!(f, "{}", pprust::path_to_string(&self.0))
```

Comment:
I do like this, and it reduces code a bunch, but paths in attrs are a lot more limited than ast/hir paths since they for example aren't name-resolved. I kind of liked having the separate type. Maybe I didn't read enough yet to have a proper opinion on this tho...